src/types.ts

```diff
@@ -30,14 +30,32 @@ export interface Options {
   usage?: boolean;
 }
 
+interface Scenario {
+  gherkinDocument: {
+    feature: {
+      name: string;
+    };
+  };
+  pickle: {
+    name: string;
+    uri: string;
+  };
+  result?: {
+    duration: { seconds: number; nanos: number };
+    status: number;
+    message: string;
+    willBeRetried?: boolean;
+  };
+}
+
 export type StepDefinitionFn = (
   name: string,
   fn: StepFn,
   opts?: StepOptions,
 ) => void;
 
 export type TearDownFn = () => Promise<void> | void;
-type HookFn = (fn: () => Promise<void> | void) => void;
+type HookFn = (fn: (scenario?: Scenario) => Promise<void> | void) => void;
```

Comment:
We should differentiate between Test Case Hooks (`After`, `Before`) and Test Case Step Hooks (`BeforeStep`, `AfterStep`) because they have different parameters.

Reply:
Awesome!